src/Parser/Hunter/Marksmanship/Modules/Talents/Sentinel.js

```diff
@@ -0,0 +1,134 @@
+import React from 'react';
+
+import Analyzer from 'Parser/Core/Analyzer';
+import Combatants from 'Parser/Core/Modules/Combatants';
+
+import SPELLS from 'common/SPELLS';
+import StatisticBox from 'Main/StatisticBox';
+import SpellIcon from 'common/SpellIcon';
+import { formatPercentage } from 'common/format';
+
+/*
+ * Your Sentinel watches over the target area for 18 sec, applying Hunter's Mark to all enemies every 6 sec.
+ */
+const MS_BUFFER = 100;
+
+const SENTINEL_DURATION = 18000;
+
+const TIME_BETWEEN_TICKS = 6000;
+
+const TICKS_PER_CAST = 4;
+
+class Sentinel extends Analyzer {
+  static dependencies = {
+    combatants: Combatants,
+
+  };
+  damage = 0;
+  applyDebuffTimestamp = null;
+  sentinelCasts = 0;
+  sentinelTicks = 0;
+  appliedDebuffsFromSentinel = 0;
+  wastedApplications = 0;
+  wastedTicks = 0;
+  lastBadTickTimestamp = 0;
+  lastSentinelCastTimestamp = 0;
+  lostTicksFromCombatEnd = 0;
+
+  on_initialized() {
+    this.active = this.combatants.selected.hasTalent(SPELLS.SENTINEL_TALENT.id);
+  }
+
+  on_byPlayer_cast(event) {
+    const spellId = event.ability.guid;
+    if (spellId !== SPELLS.SENTINEL_TICK.id && spellId !== SPELLS.SENTINEL_TALENT.id) {
```

Comment:
Considering the `if` statements that follow, this `if` is entirely superfluous.

Reply:
Makes sense - I'll just get rid of that one